src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Handles saving and loading the TaskList to/from the hard disk via a save data text file.
+ */
+public class Storage {
+    private String filepath;
+
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Loads list of Tasks from the text file specified in the Storage object's filepath.
+     *
+     * @return ArrayList of the Tasks stored in the save data file.
+     * @throws DukeException If an Exception occurs as a result of the stored save data being malformed.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try {
+            File savefile = new File(filepath);
+            Scanner saveReader = new Scanner(savefile);
+
+            while (saveReader.hasNextLine()) {
+                String savedata = saveReader.nextLine();
```

Comment:
I believe that savedata does not follow the coding standards, and it should be saveData instead.